woodwork/table_accessor.py

```diff
@@ -232,6 +241,147 @@ def _get_subset_df_with_schema(self, cols_to_include):
 
         return new_df
 
+    def _replace_nans_for_mutual_info(self, data):
+        """
+        Replace NaN values in the dataframe so that mutual information can be calculated
+
+        Args:
+            data (pd.DataFrame): dataframe to use for calculating mutual information
+
+        Returns:
+            pd.DataFrame: data with nans replaced with either mean or mode
+
+        """
+        # replace or remove null values
+        for column_name in data.columns[data.isnull().any()]:
+            column = self._schema.columns[column_name]
+            series = data[column_name]
+
+            if _is_col_numeric(column) or _is_col_datetime(column):
+                mean = series.mean()
+                if isinstance(mean, float) and not _get_ltype_class(column['logical_type']) == Double:
+                    data[column_name] = series.astype('float')
+                data[column_name] = series.fillna(mean)
+            elif _is_col_categorical(column) or _is_col_boolean(column):
+                mode = _get_mode(series)
+                data[column_name] = series.fillna(mode)
+        return data
+
+    def _make_categorical_for_mutual_info(self, data, num_bins):
+        """Transforms dataframe columns into numeric categories so that
+        mutual information can be calculated
+
+        Args:
+            data (pd.DataFrame): dataframe to use for caculating mutual information
+            num_bins (int): Determines number of bins to use for converting
+                numeric features into categorical.
+
+        Returns:
+            data (pd.DataFrame): Transformed data
```

Comment:
We should update this to match the style used in `_replace_nans_for_mutual_info` -> `pd.DataFrame: Transformed data`?

Reply:
Yeah, that makes sense to me